### PR TITLE
Add gas metering to api

### PR DESCRIPTION
**Pull request by sasurobert**
Config and gas values to API calls.

### CODE REVIEW

File: arwen/context/arwen.go [hunk 1]
```diff
@@ -7,13 +7,22 @@ import (
...
 
 	vmcommon "github.com/ElrondNetwork/elrond-vm-common"
 	"github.com/ElrondNetwork/go-ext-wasm/wasmer"
 )
 
+// gasCostFile defines a flag for the path to the preferences toml configuration file
+var gasCostFile = cli.StringFlag{
+	Name:  "gasCost",
+	Usage: "The gas cost configuration file to load",
+	Value: "../config/config.toml",
```
Comment (camilbancioiu) [inline arwen/context/arwen.go]: What does `../config/config.toml` resolve to? Will it really resolve to the `config.toml` in Arwen, or will it try to open a `config.toml` relative to the Node's executable?

File: arwen/context/arwen.go [hunk 2]
```diff
@@ -34,6 +43,7 @@ type vmContext struct {
 	blockChainHook vmcommon.BlockchainHook
 	cryptoHook     vmcommon.CryptoHook
 	imports        *wasmer.Imports
+	instance       wasmer.Instance
```
Comment (camilbancioiu) [inline arwen/context/arwen.go]: Is this intended as a step towards reusing a Wasmer instance?

File: arwen/elrondapi/bigintOps.go [hunk 3]
```diff
@@ -132,6 +132,9 @@ func bigIntgetArgument(context unsafe.Pointer, id int32, destination int32) {
 
 	value := hostContext.GetOne(destination)
 	value.Set(args[id])
+
+	gasToUse := hostContext.GasSchedule().BigIntAPICost.BigIntGetArgument
+	hostContext.UseGas(gasToUse)
```
Comment (camilbancioiu) [inline arwen/elrondapi/bigintOps.go]: `bigIntgetArgument()` should consume gas even if the received argument `id` is invalid.

File: arwen/context/arwen.go (hunk 2 above)
Comment (sasurobert) [reply arwen/context/arwen.go]: yes.

File: arwen/elrondapi/bigintOps.go (hunk 3 above)
Comment (sasurobert) [reply arwen/elrondapi/bigintOps.go]: done

File: arwen/context/arwen.go (hunk 1 above)
Comment (sasurobert) [reply arwen/context/arwen.go]: changed.

File: arwen/context/arwen.go [hunk 4]
```diff
@@ -7,13 +7,22 @@ import (
...
 	"github.com/ElrondNetwork/go-ext-wasm/wasmer"
 )
 
+// gasCostFile defines a flag for the path to the preferences toml configuration file
+var gasCostFile = cli.StringFlag{
+	Name:  "gasCost",
+	Usage: "The gas cost configuration file to load",
+	Value: "../config/config.toml",
+}
+
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply arwen/context/arwen.go]: we will have a map of opcodes, which will be read into the structure at vm creation time.

File: config/gasSchedule_test.go [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+	assert.Nil(t, err)
+
+	fmt.Printf("%+v\n", erdOp)
+
+	ethOp := &EthAPICost{}
+	err = mapstructure.Decode(gasMap, ethOp)
+	assert.Nil(t, err)
+
+	fmt.Printf("%+v\n", ethOp)
+}
```
Comment (camilbancioiu) [inline config/gasSchedule_test.go]: Optionally add a test with a `gasMap` that contains a `0` field and assert that  `CreateGasConfig()` returns an error.

**Pull request merged**